storage/simulation/rng.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Bram Gruneir (bram+code@cockroachlabs.com)
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"sort"
+	"sync"
+
+	"github.com/cockroachdb/cockroach/proto"
+)
+
+const defaultReplicationFactor = 3
+
+// rng contains all details about a single range. It also contains a map with
+// the stores that contain replicas from this range.
+type rng struct {
```

Comment:
I keep reading this name as `random number generator`. I realize you can't use the name `range`, but you could call this `Range` since it is a main package and there is no worry about it being exported since nothing can import it. If you did that, you might want to rename `node` to `Node`.